### PR TITLE
ARGO-74 - Implement recomputation execution queue

### PR DESCRIPTION
**Pull request by kaggis**
ARGO-74

### CODE REVIEW

**Comment by kaggis** [comment]
___Explanation:___
__recompute.py__ executes a specific recomputation (with _id provided by poller) and for a specific tenant

example shell call:
`./recompute.py -i "551bdd701c8a97e78635a911" -t "Tenant_FOO"`
or:
`/recompute.py --id "551bdd701c8a97e78635a911" --tenant "Tenant_FOO"`

_Note: poller should call recompute.py providing args: (id,tenant)_


___Regarding unit tests:___
Added unit tests for all implemented methods using py.test & mock:
- for utils.py
- for mongo_recompute.py
- for recompute.py

run tests by issuing in `py.test` inside `./bin` folder
_needed packages in python 2.7 enviroment: (pytest,mock)_

___Regarding documentation:___
Added comprehensive comments and doc strings to all implemented methods


please review @makistsantekidis @dpavlos @pkoro @skanct

**Comment by pkoro** [comment]
+1

Tests seem to run OK:

```
$ py.test
============================= test session starts ==============================
platform darwin -- Python 2.7.6 -- py-1.4.28 -- pytest-2.7.1
rootdir: /Users/pkoro/Repositories/ar-compute-engine/bin, inifile: 
collected 14 items 

mongo_recompute_test.py ...
recompute_test.py ......
utils_test.py .....

========================== 14 passed in 0.12 seconds ===========================
```

File: bin/upload_sync.py [hunk 1]
```diff
@@ -103,6 +103,11 @@ def main(args=None):
 
     run_cmd(cmd_call_downtimes, log)
 
+    # Call script to retrieve a json file of recomputations for the specific date/tenant from mongodb
+    cmd_mongo_recomputations = [os.path.join(argo_exec,'mongo_recompute.py'),'-d',args.date,'-t',args.tenant]
```
Comment (makistsantekidis) [inline bin/upload_sync.py]: argo_exec variable is never declared in the scope

Comment (kaggis) [inline bin/upload_sync.py]: Fixed

**Comment by makistsantekidis** [comment]
+1

**Pull request merged**